### PR TITLE
[MPLUGIN-408] helpmojo improvement

### PR DESCRIPTION
**Pull request by slawekjaranowski**
- use groupId + artifactId for calculating package name
- discover mojos only ones by descriptor mojo
- simplify in AbstractGeneratorMojo
- remove unneeded code

### CODE REVIEW

**Comment by slawekjaranowski** [comment]
Nobody wants to look?
If need more time let me know.

Failed build on jenkins on branch which was removed when PR has been created ...

**Comment by michael-o** [comment]
> Nobody wants to look? If need more time let me know.
> 
> Failed build on jenkins on branch which was removed when PR has been created ...

Massive change, takes time. It is weekend also...

**Comment by slawekjaranowski** [comment]
kindly reminder

File: maven-plugin-tools-generators/src/main/java/org/apache/maven/tools/plugin/generator/PluginDescriptorGenerator.java [hunk 1]
```diff
@@ -45,6 +42,8 @@
 import org.codehaus.plexus.util.xml.PrettyPrintXMLWriter;
 import org.codehaus.plexus.util.xml.XMLWriter;
 
+import static java.nio.charset.StandardCharsets.UTF_8;
```
Comment (michael-o) [inline maven-plugin-tools-generators/src/main/java/org/apache/maven/tools/plugin/generator/PluginDescriptorGenerator.java:45]: Where is this used in the diff?

**Comment by michael-o** [comment]
Update commit message:
- use groupId + artifactId for calculating package name
- discover mojos only once by descriptor mojo
- simplify AbstractGeneratorMojo
- remove unneeded code

File: maven-plugin-tools-generators/src/main/java/org/apache/maven/tools/plugin/generator/PluginDescriptorGenerator.java (hunk 1 above)
Comment (slawekjaranowski) [reply maven-plugin-tools-generators/src/main/java/org/apache/maven/tools/plugin/generator/PluginDescriptorGenerator.java:45]: old item, import reorder ...
used in lines: 101, 103

Comment (slawekjaranowski) [reply maven-plugin-tools-generators/src/main/java/org/apache/maven/tools/plugin/generator/PluginDescriptorGenerator.java:45]: checkstyle takes care about unused imports

**Review by michael-o — APPROVED** [review]
Except for the outstanding commit message change, I am fine with this.

**Comment by jira-importer** [comment]
Resolve #720

**Pull request merged**